userspace/libsinsp/test/public_sinsp_API/interesting_syscalls.cpp

```diff
@@ -271,59 +271,40 @@ std::set<uint32_t> ordered_sinsp_state_ppm_sc_set{
 #ifdef __NR_epoll_create1
 	PPM_SC_EPOLL_CREATE1,
 #endif
-};
+});
 
 /* This test asserts that `enforce_sinsp_state_ppm_sc` correctly retrieves
  * the `libsinsp` state ppm_sc set.
  */
 TEST(interesting_syscalls, enforce_sinsp_state_basic)
 {
-	std::unique_ptr<sinsp> inspector(new sinsp());
-	std::set<uint32_t> ordered_final_ppm_sc_set = test_utils::unordered_set_to_ordered(inspector->enforce_sinsp_state_ppm_sc());
+	auto state_ppm_sc_set = libsinsp::events::sinsp_state_sc_set();
 
-	/* Assert that the 2 sets have the same size */
-	ASSERT_EQ(ordered_sinsp_state_ppm_sc_set.size(), ordered_final_ppm_sc_set.size());
-
-	auto final = ordered_final_ppm_sc_set.begin();
-	auto matching = ordered_sinsp_state_ppm_sc_set.begin();
-
-	for(; final != ordered_final_ppm_sc_set.end(); final++, matching++)
-	{
-		ASSERT_EQ(*matching, *final);
-	}
+	ASSERT_TRUE(expected_sinsp_state_ppm_sc_set.equals(state_ppm_sc_set));
 }
 
 /* This test asserts that `enforce_sinsp_state_ppm_sc` correctly merges
  * the provided set with the `libsinsp` state set.
  */
 TEST(interesting_syscalls, enforce_sinsp_state_with_additions)
 {
-	std::unique_ptr<sinsp> inspector(new sinsp());
-	std::unordered_set<uint32_t> additional_syscalls;
-	auto ordered_ppm_sc_matching_set = ordered_sinsp_state_ppm_sc_set;
+	libsinsp::events::set<ppm_sc_code> additional_syscalls;
+	auto ppm_sc_matching_set = expected_sinsp_state_ppm_sc_set;
 
 #ifdef __NR_kill
```

Comment:
At the end of our work we should remove all these `ifdef`, we should be able to work on PPM_SC codes without considering the running architecture :tada: